inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -207,6 +208,14 @@ def submission_fulltext_download(obj, eng):
         )
 
         if pdf:
+            lb = LiteratureBuilder(source=obj.data['acquisition_source']['source'], record=obj.data)
+            lb.add_document(
+                filename,
```

Comment:
how do you ensure uniqueness here? maybe the builder should raise an exception if the key is already present.